logic/src/commonMain/kotlin/com/wire/kalium/logic/data/properties/UserPropertyRepository.kt

```diff
@@ -58,4 +62,23 @@ internal class UserPropertyDataSource(
         userConfigRepository.setReadReceiptsStatus(false)
     }
 
+    override suspend fun getTypingIndicatorStatus(): Boolean =
+        userConfigRepository.isTypingIndicatorEnabled()
+            .firstOrNull()
+            ?.fold({ false }, { it }) ?: false
+
+    override suspend fun observeTypingIndicatorStatus(): Flow<Either<CoreFailure, Boolean>> =
+        userConfigRepository.isTypingIndicatorEnabled()
+
+    override suspend fun setTypingIndicatorEnabled(): Either<CoreFailure, Unit> = wrapApiRequest {
+        propertiesApi.setProperty(PropertiesApi.PropertyKey.WIRE_TYPING_MODE, 1)
+    }.flatMap {
+        userConfigRepository.setTypingIndicatorStatus(true)
+    }
+
+    override suspend fun removeTypingIndicatorProperty(): Either<CoreFailure, Unit> = wrapApiRequest {
+        propertiesApi.deleteProperty(PropertiesApi.PropertyKey.WIRE_TYPING_MODE)
+    }.flatMap {
+        userConfigRepository.setTypingIndicatorStatus(false)
+    }
```

Comment:
```suggestion
    override suspend fun enableTypingIndicatorProperty(): Either<CoreFailure, Unit> = wrapApiRequest {
        propertiesApi.deleteProperty(PropertiesApi.PropertyKey.WIRE_TYPING_MODE)
    }.flatMap {
        userConfigRepository.setTypingIndicatorStatus(true)
    }
```

Reply:
https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/29065437/Use+case+Typing+indicator#Syncing-typing-indicator-preference-for-each-account-across-the-devices